### ISSUE TITLE
Introduce `SwitchBreak` option for the `indent` rule

### ISSUE THREAD

**Issue by Eccenux** [eslint/eslint#9895]
**What rule do you want to change?**

I want to change the `indent` rule by adding `SwitchBreak` to accompany `SwitchCase` rule.

**Does this change cause the rule to produce more or fewer warnings?**

Fewer for complying code. More for uncomplying code

**How will the change be implemented? (New option, new default behavior, etc.)?**

New option.

**Please provide some example code that this change will affect:**

```js
/*eslint indent: ["error", "tab", { "SwitchCase": 1, "SwitchBreak": 1 }]*/
function documentCodeToType (code) {
	var type;
	switch (code) {
		case 0: type  = 'Book'; break;
		case 1: type  = 'Movie'; break;
		case 2: type  = 'EDocument'; break;
		default: 
			type = 'Book';
			console.warn('documentCodeToType: unknown type: ', code);
		break;
	}
	return type;
}
```

**What does the rule currently do for this code?**

The code is invalid due to last `break` having wrong indentation.
```
Expected indentation of 3 tabs but found 2. (indent)
```

**What will the rule do after it's changed?**

The code would be valid. `break` would have expected indentation of `switch+1`.

To clarify. Currently the `SwitchBreak` is not a rule, but is actually always `SwitchCase+1`. This should be default. So effectively this should be equivalent:
```js
/*eslint indent: ["error", "tab", { "SwitchCase": 1 }]*/
/*eslint indent: ["error", "tab", { "SwitchCase": 1, "SwitchBreak": 2 }]*/
```

I can try to implement this if the rule/option is acceptable.

Also note there was a previous issue on the matter  #7270. The proposal was to allow disabling `SwitchCase` checking, but example code actually suggest the user would want `{ "SwitchCase": 1, "SwitchBreak": 1 }`.

**Comment by platinumazure** [comment]
Hi @Eccenux, thanks for the issue (and for taking the time to fill out our rule change proposal template)!

I'm on the fence about this, due to `indent` already being really complicated and not being sure if a lot of people would really benefit from this change. I'm not going to downvote the proposal, but I'm not yet convinced that we should support this.

**Comment by not-an-aardvark** [comment]
Thanks for your interest in improving ESLint. Unfortunately, it looks like this issue didn't get consensus from the team, so I'm closing it. We define [consensus](http://eslint.org/docs/maintainer-guide/issues.html#consensus) as having three :+1:s from team members, as well as a team member willing to champion the proposal. This is a high bar by design -- we can't realistically accept and maintain every feature request in the long term, so we only accept feature requests which are useful enough that there is consensus among the team that they're worth adding.

Since ESLint is pluggable and can load custom rules at runtime, the lack of consensus among the ESLint team doesn't need to be a blocker for you using this in your project, if you'd find it useful. It just means that you would need to [implement the rule](http://eslint.org/docs/developer-guide/working-with-rules) yourself, rather than using a bundled rule that is packaged with ESLint.
